### PR TITLE
Fabric: auto default

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Part of https://github.com/Lightning-AI/lightning/issues/10606

cc @justusschock @awaelchli @borda @carmocca

### CODE REVIEW

File: docs/source-pytorch/fabric/fundamentals/launch.rst [hunk 1]
```diff
@@ -47,10 +47,10 @@ This is essentially the same as running ``python path/to/your/script.py``, but i
       itself and are expected to be parsed there.
 
     Options:
-      --accelerator [cpu|gpu|cuda|mps|tpu]
+      --accelerator [auto|cpu|gpu|cuda|mps|tpu]
                                       The hardware accelerator to run on.
-      --strategy [ddp|dp|deepspeed]   Strategy for how to run across multiple
-                                      devices.
+      --strategy [auto|ddp|dp|deepspeed]
+                                      Strategy for how to run across multiple devices.
```
Comment (awaelchli) [inline docs/source-pytorch/fabric/fundamentals/launch.rst:53]: The cli can't parse auto at the moment. The code would have to be refactored. I suggest leaving this for another PR.

File: tests/tests_fabric/plugins/precision/test_amp_integration.py [hunk 2]
```diff
@@ -68,7 +68,9 @@ def after_backward(self, model):
     ],
 )
 def test_amp(accelerator, precision, expected_dtype):
-    fabric = MixedPrecisionBoringFabric(accelerator=accelerator, precision=precision)
+    # TODO: devices>1 fails with:
+    # DDP expects same model across all ranks, but Rank 0 has 2 params, while rank 1 has inconsistent 1 params
+    fabric = MixedPrecisionBoringFabric(accelerator=accelerator, precision=precision, devices=1)
```
[the remark this answers is not in the document]
Comment (carmocca) [reply tests/tests_fabric/plugins/precision/test_amp_integration.py:73]: It is. I also was confused about this, hence why I left it as a TODO
https://dev.azure.com/Lightning-AI/lightning/_build/results?buildId=138771&view=logs&j=3f274fac-2e11-54ca-487e-194c91f3ae9f&t=7ba6ccc2-87f1-5d8a-85e1-a52b03435dd6&l=352

File: docs/source-pytorch/fabric/fundamentals/launch.rst (hunk 1 above)
Comment (carmocca) [reply docs/source-pytorch/fabric/fundamentals/launch.rst:53]: I'll revert this

File: tests/tests_fabric/plugins/precision/test_amp_integration.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (awaelchli) [reply tests/tests_fabric/plugins/precision/test_amp_integration.py:73]: Okay, that's insane. Could you open an issue and assign me?

[the remark this answers is not in the document]
Comment (carmocca) [reply tests/tests_fabric/plugins/precision/test_amp_integration.py:73]: I'll open it when it gets merged!

**Review by awaelchli — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Pull request merged**